src/rust/ide/lib/view/src/graph_editor/component/edge.rs

```diff
@@ -39,118 +28,550 @@ const LINE_SHAPE_WIDTH   : f32 = LINE_WIDTH + 2.0 * PADDING;
 const LINE_SIDE_OVERLAP  : f32 = 1.0;
 const LINE_SIDES_OVERLAP : f32 = 2.0 * LINE_SIDE_OVERLAP;
 const LINE_WIDTH         : f32 = 4.0;
+const ARROW_SIZE_X       : f32 = 20.0;
+const ARROW_SIZE_Y       : f32 = 20.0;
+
+const HOVER_EXTENSION    : f32 = 10.0;
+
 const MOUSE_OFFSET       : f32 = 2.0;
 const NODE_PADDING       : f32 = node::SHADOW_SIZE;
-const PADDING            : f32 = 4.0;
+const PADDING            : f32 = 4.0 + HOVER_EXTENSION;
 const RIGHT_ANGLE        : f32 = std::f32::consts::PI / 2.0;
 
+// TODO: Maybe find a better name.
+const MIN_SOURCE_TARGET_DIFFERENCE_FOR_Y_VALUE_DISCRIMINATION : f32 = 45.0;
+
 const INFINITE : f32 = 99999.0;
 
+const HOVER_COLOR : color::Rgba = color::Rgba::new(1.0,0.0,0.0,0.000_001);
+
+
+
+fn up() -> Vector2<f32> {
+    Vector2(1.0,0.0)
+}
+
+fn point_rotation(point:Vector2<f32>) -> nalgebra::Rotation2<f32> {
+    nalgebra::Rotation2::rotation_between(&point,&up())
+}
+
```

Comment:
I know it was my change, but I did not clean my code. This should be in a section imo